h2/src/main/org/h2/mvstore/type/ByteArrayDataType.java

```diff
@@ -43,4 +44,9 @@ public byte[] read(ByteBuffer buff) {
     public byte[][] createStorage(int size) {
         return new byte[size][];
     }
+
+    @Override
+    public int compare(byte[] one, byte[] two) {
+        return Arrays.compare(one, two);
```

Comment:
H2 uses `Arrays.compareUnsigned()` for binary string data types, but octets in SQL and bytes in Java aren't exactly the same thing.

Anyway, it is possible to write a custom data type for MVStore with any comparison rules.

Reply:
Either way I could care less, it is mostly for benchmark testing and comparison with other stores